db/migrate/20150819150242_add_more_fields_to_project_details.rb

```diff
@@ -0,0 +1,133 @@
+class AddMoreFieldsToProjectDetails < ActiveRecord::Migration
+  def up
+    execute <<-SQL
+      drop function if exists public.published_states();
+      CREATE FUNCTION public.published_states() RETURNS text[]
+        LANGUAGE sql STABLE SECURITY DEFINER
+          AS $$
+            SELECT '{"online", "waiting_funds", "failed", "successful"}'::text[];
+          $$;
+
+      drop function if exists public.is_published(projects);
+      create function public.is_published(projects) returns boolean
+        language sql stable SECURITY DEFINER
+        as $$
+          select true where $1.state = ANY(public.published_states());
+        $$;
+
+      drop function if exists public.is_expired(projects);
+      CREATE FUNCTION public.is_expired(projects) RETURNS boolean
+        LANGUAGE sql STABLE SECURITY DEFINER
+          AS $$
+            SELECT (current_timestamp > $1.expires_at);
+          $$;
+
+      drop function if exists public.open_for_contributions(projects);
+      CREATE FUNCTION public.open_for_contributions(projects) RETURNS boolean
+        LANGUAGE sql STABLE SECURITY DEFINER
+          AS $$
+            SELECT (not $1.is_expired AND $1.state = 'online')
+          $$;
+
+      drop function if exists public.remaining_time_interval(projects);
+      CREATE FUNCTION public.remaining_time_interval(projects) RETURNS interval
+        LANGUAGE sql STABLE SECURITY DEFINER
+          AS $$
+            select ($1.expires_at - current_timestamp)::interval
+          $$;
+
+      drop function if exists public.remaining_time_json(projects);
+      CREATE FUNCTION public.remaining_time_json(projects) RETURNS json
```

Comment:
Let's try to unify this function and the one we have in the ```ProjectDecorator``` called ```time_to_go```. It's better to have a standard for this string and also not duplicate logic